commands/remove.js

```diff
@@ -8,24 +8,38 @@ module.exports = {
 	groupArgs: true,
 	execute: async (bot, msg, args, cfg) => {
 		if(!args[0]) return bot.cmds.help.execute(bot, msg, ["remove"], cfg);
-		
+
 		//check arguments
 		if(args[0] == "*") {
-			try {
-				await bot.send(msg.channel, `Warning: This will remove ALL of your ${cfg.lang}s. Reply 'yes' to continue or anything else to cancel.`);
-				let response = await bot.waitMessage(msg);
-				if(response.content.toLowerCase() != "yes") return "Canceling operation.";
-			} catch(e) {
-				if(e == "timeout") return "Response timed out. Canceling.";
-				else throw e;
-			}
+			if ((await bot.db.members.getAll(msg.author.id)).length == 0) return "You don't have anything to remove.";
+			let confirm = await bot.confirm(msg, `Warning: This will remove ALL of your ${cfg.lang}s. Reply 'yes' to continue or anything else to cancel.`);
+			if (confirm !== true) return confirm;
 			await bot.db.members.clear(msg.author.id);
 			return `All ${cfg.lang}s removed.`;
 		}
-		let name = args.join(" ");
-		let member = await bot.db.members.get(msg.author.id,name);
-		if(!member) return "Could not find " + cfg.lang + " with that name registered under your account.";
-		
+		else if (args.length == 1) {
+			let name = args.join(" ");
+			let member = await bot.db.members.get(msg.author.id,name);
+			if(!member) return "Could not find " + cfg.lang + " with that name registered under your account.";
+		} else {
+			let removedMessage = `${proper(cfg.lang)}s removed:`
+			let notRemovedMessage = `${proper(cfg.lang)}s not found:`
+			let baseLength = 2000 - (removedMessage.length + notRemovedMessage.length)
+			let rOriginalLength = { removedMessage: removedMessage.length, notRemovedMessage: notRemovedMessage.length, }
+
+			for await (let arg of args) {
+				tup = await bot.db.members.get(msg.author.id, arg);
+				if (tup) {
+					await bot.db.members.delete(msg.author.id, arg);
+					if ((removedMessage.length + notRemovedMessage.length + arg.length) < baseLength) removedMessage += ` '${arg}'`; else removedMessage += " (...)";
```

Comment:
this should be compressed into one query (using unnest)

Reply:
Not sure how that would work here

Reply:
two queries might work instead, one for selecting all tuppers in a list and then the other for updating just those tuppers, instead of doing them one at a time

Reply:
Oh yeah that could work